src/apps/common/submodules/navigationWizard/navigationWizard.js

```diff
@@ -0,0 +1,206 @@
+		_ = require('lodash'),
+		monster = require('monster');
+
+	var navigationWizard = {
+		requests: {
+
+		},
+
+		subscribe: {
+			'common.navigationWizard.render': 'navigationWizardRender'
+		},
+
+		navigationWizardRender: function(args) {
+			var self = this,
+				container = args.container,
+				currentStep = 0,
+				layout = $(self.getTemplate({
+					name: 'layout',
+					data: {
+						title: args.title,
+						steps: args.steps,
+						doneButton: args.doneButton ? args.doneButton : ''
+					},
+					submodule: 'navigationWizard'
+				}));
+
+			if (container) {
+				self.navigationWizardBindEvents($.extend({ template: layout }, args));
+
+				container
+					.empty()
+					.append(layout);
+
+				self.generateTemplate(currentStep, args);
+			} else {
+				throw new Error('A container must be provided.');
+			}
+		},
+
+		navigationWizardBindEvents: function(args) {
+			var self = this,
+				thisArg = args.thisArg,
+				template = args.template,
+				currentStep = 0;
+
+			self.setSelected(currentStep, args);
+			self.generateTemplate(currentStep, args);
+
+			//Clicking next on the menu item
+			template
+				.find('#next')
+					.on('click', function(event) {
+						event.preventDefault();
+
+						var result = self.utilForTemplate(currentStep, args);
+
+						if (result.valid === true) {
+							//display checkbox
+							template
+								.find('i[data-id="' + currentStep + '"]')
+								.removeClass('hide-check');
+
+							template
+								.find('li[data-id="' + currentStep + '"]')
+								.addClass('done');
+
+							//merge data
+							args = _.merge({}, args, {
+								data: result.data
+							});
+
+							currentStep += 1;
+							self.setSelected(currentStep, args);
+							self.generateTemplate(currentStep, args);
+						}
+					});
+
+			//Clicking next on the menu item
+			template
+				.find('#done')
+					.on('click', function(event) {
+						event.preventDefault();
+
+						var result = self.utilForTemplate(currentStep, args);
+
+						if (result.valid === true) {
+							thisArg[args.done](args);
+						}
+					});
+
+			//Clicking on the menu item
+			template
+				.find('.back')
+					.on('click', function(event) {
+						event.preventDefault();
+
+						currentStep -= 1;
+
+						self.setSelected(currentStep, args);
+						self.generateTemplate(currentStep, args);
+					});
+
+			//Clicking on cancel
+			template
+				.find('#cancel')
+					.on('click', function(event) {
+						event.preventDefault();
+
+						thisArg[args.cancel](args);
+					});
+
+			template
+				.find('#clear')
+					.on('click', function(event) {
+						event.preventDefault();
+
+						var step = args.steps[currentStep],
+							formattedData = _.merge({}, args, {
+								data: step.default
+							});
+
+						//re-render template with default values
+						self.setSelected(currentStep, formattedData);
+						self.generateTemplate(currentStep, formattedData);
+					});
+		},
+
+		setSelected: function(currentStep, args) {
```

Comment:
Technically, common controls are submodules so each method should be prefixed by the submodule name:
```suggestion
		navigationWizardSetSelected: function(currentStep, args) {
```

(don't forget to update the method calls as well)

Reply:
I made the changes you requested, thanks!